cpu/stm32_common/periph/flash_common.c

```diff
@@ -56,3 +57,23 @@ void _lock(void)
         CNTRL_REG |= CNTRL_REG_LOCK;
     }
 }
+
+void _wait_for_pending_operations(void)
+{
+    if ((FLASH->SR & FLASH_SR_BSY) == FLASH_SR_BSY) {
+        DEBUG("[flash-common] waiting for any pending operation to finish\n");
+        while (FLASH->SR & FLASH_SR_BSY) {}
+    }
+    else {
```

Comment:
Comparing with master, the `else` block is an addition and it fails on stm32f1 which doesn't provide the `FLASH_SR_WRPERR` flag.
There are 2 possibilities:
- provide the flash error handling in a separate PR
- keep it, but put it in its own commit and fix the stm32f1 case.

Reply:
Ok, I'll go for option 1, I'll look a little deeper into the error handling and provided it in a future PR. For now I'll just remove it.